### PR TITLE
chore(deps-dev): bump typedoc to 0.23.23

### PR DESCRIPTION
**Pull request by MYoung25**
### Issue
https://github.com/aws/aws-sdk-js-v3/issues/4134.  This is a simplified version of PR https://github.com/aws/aws-sdk-js-v3/pull/4458

### Description
Upgrade TypeDoc to 0.23.23 and fix identified a11y issues.

### Testing
`yarn generate-clients && yarn build:all && yarn build:docs` 

### Additional context
Internal information in JS-3393, JS-4030, and CR-85945847

This PR requires merging CR-85945847 and https://github.com/awslabs/smithy-typescript/pull/699

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: typedoc.client.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
-  "exclude": ["**/node_modules/**", "**/protocols/*.ts", "**/endpoints.ts", "**/ruleset.ts"],
```
Comment (kuhe) [inline typedoc.client.json:2]: `**/ruleset.ts` needs to still be included

Comment (MYoung25) [reply typedoc.client.json:2]: Addressed in ce91847e4567abe5aaa01707cc7a347694f12a18

**Review by milesziemer — APPROVED** [review]

File: clients/client-accessanalyzer/typedoc.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "../../typedoc.client.json"
+  "extends": ["../../typedoc.client.json"],
+  "entryPoints": ["src/index.ts"],
+  "out": "docs",
+  "readme": "README.md"
```
Comment (trivikr) [inline clients/client-accessanalyzer/typedoc.json:5]: The `out` and `readme` configuration may be used from root typedoc.json
We might take up this work as a follow-up.

File: private/aws-client-api-test/package.json [hunk 3]
```diff
@@ -26,7 +25,7 @@
     "@types/node": "^12.7.5",
     "concurrently": "7.0.0",
     "downlevel-dts": "0.10.1",
-    "typedoc": "0.19.2",
+    "typedoc": "^0.23.23",
```
Comment (trivikr) [inline private/aws-client-api-test/package.json:28]: Suggestion: Either use pinned dependency of typedoc, or remove it entirely for private clients.
```suggestion
    "typedoc": "0.23.23",
```

**Review by trivikr — CHANGES_REQUESTED** [review]

**Review by trivikr — APPROVED** [review]

**Pull request merged**